### DISCUSSION TITLE
2007/03/29/%E7%B3%BB%E7%BB%9F%E9%9C%80%E8%A6%81%E4%BB%AA%E5%99%A8%E6%9C%89%E5%8F%AF%E4%BA%92%E6%8D%A2%E6%80%A7%EF%BC%8C%E4%BD%86%E6%98%AF%E6%B2%A1%E6%9C%89%E7%9B%B8%E5%BA%94%E7%9A%84-ivi-%E7%B1%BB%E9%A9%B1

### DISCUSSION THREAD

**Discussion by giscus** [ruanqizhen/blog#147] [Announcements]
# 2007/03/29/%E7%B3%BB%E7%BB%9F%E9%9C%80%E8%A6%81%E4%BB%AA%E5%99%A8%E6%9C%89%E5%8F%AF%E4%BA%92%E6%8D%A2%E6%80%A7%EF%BC%8C%E4%BD%86%E6%98%AF%E6%B2%A1%E6%9C%89%E7%9B%B8%E5%BA%94%E7%9A%84-ivi-%E7%B1%BB%E9%A9%B1

问题(Martin)：

https://blog.qizhen.xyz/2007/03/29/%E7%B3%BB%E7%BB%9F%E9%9C%80%E8%A6%81%E4%BB%AA%E5%99%A8%E6%9C%89%E5%8F%AF%E4%BA%92%E6%8D%A2%E6%80%A7%EF%BC%8C%E4%BD%86%E6%98%AF%E6%B2%A1%E6%9C%89%E7%9B%B8%E5%BA%94%E7%9A%84-ivi-%E7%B1%BB%E9%A9%B1

**Comment by ruanqizhen** [comment]
之前博客下的留言：

方
2011年04月4日 13:20 编辑
您好，今天刚收到您的书，目前只看的了序和前言，感觉不错 。我是一个学生，目前在做一个项目，测试相关的参数，发现LabVIEW是入门易精通难的，希望能与您有交流的机会！！！

回复
ruanqizhen
2011年04月4日 18:31 编辑
多多交流

回复
Tom Lu
2007年04月8日 14:33 编辑
以前接触过一个程序采用Plug-in的方式来处理这个问题的，就是两种仪器的控制用不同的vi但接口是一样的，可以通过配置决定调用那一个仪器的vi。

回复
Unknown
2007年04月3日 17:01 编辑
感谢回复以及大家的谈论.高杰的办法也是一个不错的方法.
我现在打算用VI server的方法来实现一些仪器的互换,也就阮回复的那样.
工作还在进行中.
另外祝新婚快乐.
 
我还看了你写的关于 荨麻疹 经历, 感触很深,因为今年过年时我的妈妈也得了这种病,但当时对它不了解,以为就是
一般的皮肤过敏,吃点药就好了.后来才得知这种病要是变成慢性的会很折麽人的. 所以现在很担心,也后悔为什么没
在家多陪陪老妈.
 
健康.
 

回复
Unknown
2007年04月3日 17:01 编辑
感谢回复以及大家的谈论.高杰的办法也是一个不错的方法.
我现在打算用VI server的方法来实现一些仪器的互换,也就阮回复的那样.
工作还在进行中.
另外祝新婚快乐.
 
我还看了你写的关于 荨麻疹 经历, 感触很深,因为今年过年时我的妈妈也得了这种病,但当时对它不了解,以为就是
一般的皮肤过敏,吃点药就好了.后来才得知这种病要是变成慢性的会很折麽人的. 所以现在很担心,也后悔为什么没
在家多陪陪老妈.
 
健康.
 

回复
Qizhen
2007年04月2日 16:21 编辑
回 怒放的生命关于具备什么样的条件才能进一个公司这很难说的。每个面试官可能都有不同的观点，公司之间大概差距更大。
现在LabVIEW的应用也不是很普及，我们公司招来的新员工大多都是没有用过LabVIEW的。所以是否会LabVIEW可能不想你想得那么重要。
我个人觉得，当前的学习和工作成绩可能对将来的求职影响最大吧。比如说，如果你现在成绩总是在系里前几名，实践能力也不差，心目中有五六家比较理想的公司，都去面试一下，肯定会被其中的某系公司看中的。能把现在的事情做得很优秀，就不用太担心以后的工作了。

回复
Qizhen
2007年04月2日 15:55 编辑
自己开发 IVI Class Driver 不但麻烦，而且其他公司也不见得认同你的标准。开发出来的 Class Driver 只能配合自己写的 Specfic Driver。
所以我觉得高杰的办法也不错，比开发新的 Class Driver 效率高。另外我以前还用过一个方法，适合简单的一起操作。如果一个系统之用到很简单的几个一起操作，比如：“打开”、“配置”、“读数据”“关闭”。可以针对你用到的所以得仪器都编写四个这样的函数（VI），保持参数的个数和类型一致。在应用程序中根据仪器的不同动态的调用相应的每个仪器的函数。这种方法也让系统具备了可互换性，并且实现起来也相对简单。

回复
清扬
2007年04月2日 12:46 编辑
      阮先生,我想问下,如果想进NI的话,除了精通labview之外,还需要懂什么呢?我现在还在上研究生,非常喜欢labview,希望能进一个象NI这样的大公司,能和您这样的真正热爱技术的人共事,希望您能给点指导.现在在学校很迷茫,虽然还有一年就毕业了,可还不知道自己究竟需要具备什么样的素质才能被理想中的公司录用.我曾经看过您的一篇关于招聘的文章,我想前辈的经验一定会更有利于我们职业规划的.谢谢~

回复
杰
2007年03月29日 22:55 编辑
自己开发标准而全面的某一种类的IVI Class Driver可能真的麻烦了一点。甚至于有的时候，某些仪器不支持IVI。不过其实在真正的工程应用中，还是可以方便的实现的。
所谓IVI就是具有相同功能的仪器在系统中的硬件互换而不影响软件。（我自己的理解，可能有些许偏差。呵呵。）
那么，很简单的做一个小的ACESS甚至于一个TXT文件把需要用到的Program command分类的全部写进去。程序初始化的时候用*IDN识别硬件，作为索引，在软件中需要用到的地方找出相应的command发给相应的硬件即可。
因为不同的硬件厂商虽然对仪器的操作命令不同，但*IDN？这样的通用命令是都支持的。
我开发的一套ATS里面就是这么干的，呵呵。里面大概能用3种不同的示波器、5种不同的DC POWER，2种不同的网络分析仪。其实真正在工程中应用之后，都做成小模块了，会发现即使再多种类的不同厂家的相同类的仪表，要把软件做成通用的，还是很方便的。看一下仪器的manual，在记录command的文件里面添加一行数据即可。不过我那个是用VB写的，LV我还是新手上路。但我觉得这种编程的思想还是通用。
可能俺的这种方法太老土，不过还是有效。见笑了。

回复
